libobs/obs.c

```diff
@@ -634,17 +634,17 @@ static bool obs_init_data(void)
 	pthread_mutex_init_value(&obs->data.displays_mutex);
 	pthread_mutex_init_value(&obs->data.draw_callbacks_mutex);
 
-	if (pthread_mutex_init_recursive(&data->sources_mutex) != 0)
+	if (pthread_rwlock_init(&data->sources_rwlock, NULL) != 0)
```

Comment:
Since we're losing recursive mutex locking functionality here, any calls that risk requiring an upgrade from read-only to rw need to assume rw. Unfortunately, this can be in a lot of places. The most common I can gather is anywhere `obs_source_release()` is called, that might trigger a destroy which requires holding a rw lock.